planner/core/planbuilder.go

```diff
@@ -1424,6 +1425,19 @@ func (b *PlanBuilder) buildDDL(node ast.DDLNode) Plan {
 				table:     v.ReferTable.Name.L,
 			})
 		}
+	case *ast.CreateViewStmt:
+		plan, err := b.Build(v.Select)
```

Comment:
After thinking, we do not need to record the columnName of the underlying table, e.g.
``` go
create table t (a int, b int);
create view v(c, d) as select a, b from t;
```
We'll build a `Projection` upon `buildSelect(select a, b from t)`.
Since there is a one-to-one correspondence between `the offsets of the columns in Projection` and `the offsets of the columns in the schema of the underlying table`.
We can assign the `childSelect.schema.Column[i]` to `Projection.Exprs[i]` directly without knowning `the relation of the names`.
[This function](https://github.com/pingcap/tidb/blob/master/planner/core/logical_plan_builder.go#L698) can be referred.

Reply:
We can assign the `schema.ColumnAsNames` to `CreateViewStmt.Cols`.
And the p.SetSchema() can be removed.

Reply:
This check also should be added here:
if v.Cols != nil && len(v.Cols) != len(schema.Len()) {
    return err
}